src/views/ListOfBooks.jsx

```diff
@@ -33,18 +35,36 @@ const ListOfBooks = () => {
     });
   },[]);
 
+  // Filter books by subject
+  const subjects = books?.flatMap((book) => book.subjects?.map((subject) => subject.name));
+
+  const onlyUnique = (value, index, self) => {
+    return self.indexOf(value) === index;
+  }
+
+  // Get unique subjects to display in the select option
+  const subjectsList = subjects?.filter(onlyUnique);
+
   return (
    <div className="books-list">
+     <SubjectFilter
+        subjectsList={subjectsList}
+        activeSubject={activeSubject}
+        setActiveSubject={setActiveSubject}
+      />
       {books && books.map((book) => {
-        return (
-          <div className="books-container" key={book.id}>
-            <Link to={book.valid && `/book/${book.id}`} state={{ book }}>
-              <h3 className={book.valid ? "" : "not-valid-books-title"}>{book.displayTitle}</h3>
-              <img className={book.valid ? "valid-books-img" : "not-valid-books-img"} src={book.url} alt={book.displayTitle} />
-            </Link>
-          </div>
-        )
-      })}
+        if (activeSubject === "" || book.subjects.some((subject) => subject.name === activeSubject)) {
```

Comment:
In many cases, `map` functions can be written as a lambda expression to cut down on the noise. And `if () then {} else {}` can be replaced with ternary expressions.

Reply:
I am not sure I fully understand why I should replaced the if/else statement with ternary expression (in that specific case) as for me it will not be as lisible ?

I removed the "return null" as it was not mandatory

Reply:
In this case, I agree with you, a ternary might be less legible… but also because the `map` is being used as both `map` and `filter`.

It's a good practice to separate the logic of filtering from the micro-logic of presenting a list item depending on its data.

Here, you could use a `useMemo` hook to filter the list of books based on user selection. The `map` then becomes much simpler.